src/lib.rs

```diff
@@ -42,19 +49,20 @@ pub fn run(args: Args) {
             .capture_index_for_name(capture_name)
             .expect(&format!("Unknown capture name: `{}`", capture_name))
     });
+
     enumerate_project_files(&*supported_language)
         .par_iter()
-        .flat_map(|project_file_dir_entry| {
-            get_results(
-                &query,
-                project_file_dir_entry.path(),
-                capture_index,
-                language,
-            )
+        .for_each(|project_file_dir_entry| {
+            let mut printer = grep::printer::Standard::new_no_color(io::stdout());
+            let path = project_file_dir_entry.path();
+            let matches = get_results(&query, path, capture_index, language);
+
+            let matcher = TreeSitterMatcher::new(matches);
+
+            Searcher::new()
+                .search_path(&matcher, path, printer.sink_with_path(&matcher, path))
+                .unwrap();
```

Comment:
I don't think this is really getting us much of anything, since it's really meant to deal with incremental searching in a smart way, but it seemed like the easiest way to use the printer, given the printer's interface.